src/System.Private.CoreLib/src/System/AppContextConfigHelper.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+
+namespace System
+{
+    internal static class AppContextConfigHelper
+    {
+        internal static int GetConfig(string configName, int defaultValue)
+        {
+            object config = AppContext.GetData(configName);
+            switch (config)
+            {
+                case string str:
+                    if (str.StartsWith("0x"))
+                    {
+                        return Convert.ToInt32(str, 16);
```

Comment:
Should catch all exceptions and fall back to the default value in case of exception, same elsewhere